src/miner_lora.erl

```diff
@@ -903,6 +903,32 @@ retry_with_rx2(HlmPacket0, From, State) ->
     send_packet(Payload, TS, ChannelSelectorFun, DataRate, Power, true, HlmPacket1, From, State).
 
 -spec maybe_update_reg_data(State :: state()) -> state().
+maybe_update_reg_data(#state{reg_domain_confirmed=true, chain=undefined} = State) ->
+    %% don't have chain, do nothing
+    State;
+maybe_update_reg_data(#state{reg_domain_confirmed=true, chain=Chain} = State) when Chain /= undefined ->
+    case application:get_env(miner, region_override, undefined) of
+        undefined ->
+            %% region is confirmed without region_override
```

Comment:
I don't understand the logic here at all, we've already checked this in init and set reg_domain_confirmed=true

Reply:
Oh I see, disregard.